src/components/metadata/metadata.test.js

```diff
@@ -323,51 +353,75 @@ describe('MetaData', () => {
         return setup.mount(<MetaData visible={true} />, {
           beforeLayoutActions: [() => toggleTypeDisabled('parameters', false)],
           afterLayoutActions: [
-            () => {
-              // Click the expected node
-              return toggleNodeClicked(props.nodeId);
-            },
+            // Click the expected node
+            () => toggleNodeClicked(props.nodeId),
+            //simulating loadNodeData in node.js
+            () =>
+              addNodeMetadata({ id: props.nodeId, data: props.mockMetadata }),
           ],
         });
       };
       it('shows the node type as an icon', () => {
-        const wrapper = mount({ nodeId: rabbitParamsNodeId });
+        const wrapper = mount({
+          nodeId: rabbitParamsNodeId,
+          mockMetadata: node_parameters,
+        });
         expect(
           rowIcon(wrapper).hasClass('pipeline-node-icon--icon-parameters')
         ).toBe(true);
       });
 
       it('shows the node name as the title', () => {
-        const wrapper = mount({ nodeId: rabbitParamsNodeId });
+        const wrapper = mount({
+          nodeId: rabbitParamsNodeId,
+          mockMetadata: node_parameters,
+        });
         expect(textOf(title(wrapper))).toEqual(['Params:rabbit']);
       });
 
       it('shows the node type as text', () => {
-        const wrapper = mount({ nodeId: rabbitParamsNodeId });
+        const wrapper = mount({
+          nodeId: rabbitParamsNodeId,
+          mockMetadata: node_parameters,
+        });
         const row = rowByLabel(wrapper, 'Type:');
         expect(textOf(rowValue(row))).toEqual(['parameters']);
       });
 
       it('shows the node filepath', () => {
-        const wrapper = mount({ nodeId: rabbitParamsNodeId });
+        const wrapper = mount({
+          nodeId: rabbitParamsNodeId,
+          mockMetadata: node_parameters,
+        });
         const row = rowByLabel(wrapper, 'File Path:');
         expect(textOf(rowValue(row))).toEqual(['-']);
       });
 
       it('does not display the node parameter when it is an empty object', () => {
```

Comment:
if we were to go for the new approach, this wording will now need to be revisited